models/RdfImporter.php

```diff
@@ -44,6 +104,116 @@ function ($resource)
+
+        $importedCount = 0;
+
+        try {
+            $strategy = $this->readStrategyFromConfig();
+        } catch (InconsistencyConfigException $e) {
+            $report->add(
+                new Report(
+                    Report::TYPE_ERROR,
+                    self::class . " configured incorrectly"
+                )
+            );
+
+            $report->setType(Report::TYPE_ERROR);
+            $report->setMessage('Data import failed');
+
+            $this->getLogger()->logError($e->getMessage());
+
+            return $report;
+        }
+
+        foreach ($data as $subjectUri => $propertiesValues) {
+            $loginProperty = current($propertiesValues[UserRdf::PROPERTY_LOGIN]);
+            $login = $loginProperty['value'];
+
+            $isDuplicated = $this->isDuplicated($login);
+
+            $duplicationReport = null;
+
+            if ($isDuplicated && in_array($strategy, self::NOT_IMPORTABLE_STRATEGIES, true)) {
+                $message = "User `%s` was duplicated.";
+                $report->add(new Report(Report::TYPE_WARNING, sprintf($message, $login)));
+
+                if ($strategy === self::OPTION_STRATEGY_FAIL_ON_DUPLICATE) {
+                    $report->add(
+                        new Report(
+                            Report::TYPE_ERROR,
+                            'Since the `Fail on duplicate` strategy was chosen, import will now fail'
+                        )
+                    );
+
+                    break;
+                }
+
+                if ($strategy === self::OPTION_STRATEGY_SKIP_ON_DUPLICATE) {
+                    $report->add(
+                        new Report(
+                            Report::TYPE_WARNING,
+                            'Since the `Skip on duplicate` strategy was chosen, import will now skip this user, without importing it'
+                        )
+                    );
+
+                    continue;
+                }
+            }
+
+            $resource = new core_kernel_classes_Resource($map[$subjectUri]);
+            $subreport = $this->importProperties($resource, $propertiesValues, $map, $class);
+            $report->add($subreport);
+
+            if (!$subreport->containsError()) {
+                $importedCount++;
+            }
+
+            if ($isDuplicated && $strategy === self::OPTION_STRATEGY_IMPORT_ON_DUPLICATE){
+                $report->add(
+                    new Report(
+                        Report::TYPE_WARNING,
+                        'Since the `Import on duplicate` strategy was chosen, import will import the user, but behaviour is unpredicted'
+                    )
+                );
+            }
+        }
+
+        if ($importedCount > 0 && !$report->containsError()) {
+            $report->setType(Report::TYPE_SUCCESS);
+            $report->setMessage(__('Data imported successfully'));
+        }
+
+        return $report;
+    }
+
+    private function isDuplicated(string $login): bool
```

Comment:
This will be called for each user being imported, should we have concerns regarding performance here?

Reply:
Yes and no. This importer is being called under taskQueue task, so it won't give major performance issues